custom_build.sh

```diff
@@ -0,0 +1,191 @@
+readonly MARLIN_ARTIFACT_DIRECTORY=${ARTIFACT_DIRECTORY}/marlin/
+
+function help(){
+	if [ "$#" -ne 6 ] ; then
+		echo "Usage: $0 [docker image tag] [geoserver version] [geoserver master version] [datadir| nodatadir] [pull|no pull];"
+		echo "";
+		echo "[docker image tag] :          the tag to be used for the docker iamge ";
+		echo "[geoserver version] :         the release version of geoserver to be used; you can set it to master if you want the last release";
+		echo "[geoserver master version] :  if you use the master version for geoserver you need to set it to the numerical value for the next release;"
+		echo "                              if you use a released version you need to put it to the release number";
+		echo "[datadir| nodatadir]:         if this parameter is equal to nodatadir the datadir is not burned in the docker images ";
+		echo "[pull|no pull]:               docker build use always a remote image or a local image";
+		exit 1;
+	fi
+}
+
+function clean_up_directory() {
+	rm -rf ${1}/*
+}
+
+function download_from_url_to_a_filepath {
+	URL=${1}
+	FILE_PATH=${2}
+	FILE_DOWNLOADED=$(basename "${FILE_PATH}" )
+	if [ -f "${FILE_PATH}" ]; then
+		rm -f "${FILE_PATH}"
+	fi
+	if [ ! -f "${FILE_PATH}" ]; then
+		curl -L "${URL}" --output "${FILE_PATH}"
+		echo "* ${FILE_DOWNLOADED} artefact dowloaded *"
+	else
+		echo "* ${FILE_DOWNLOADED} artefact already dowloaded *"
+	fi
+}
+
+function download_plugin()  {
+	TYPE=${1}
+	PLUGIN_NAME=${2}
+
+	case ${GEOSERVER_VERSION} in
+		"${GEOSERVER_MASTER_VERSION::-2}.x")
+		PLUGIN_FULL_NAME=geoserver-${GEOSERVER_VERSION::-2}-SNAPSHOT-${PLUGIN_NAME}-plugin.zip
+		PLUGIN_ARTIFACT_URL=${BASE_BUILD_URL}/${GEOSERVER_VERSION}/${TYPE}-latest/${PLUGIN_FULL_NAME}
+		;;
+
+		"master")
+		PLUGIN_FULL_NAME=geoserver-${GEOSERVER_MASTER_VERSION::-2}-SNAPSHOT-${PLUGIN_NAME}-plugin.zip
+		PLUGIN_ARTIFACT_URL=${BASE_BUILD_URL}/${GEOSERVER_VERSION}/${TYPE}-latest/${PLUGIN_FULL_NAME}
+		;;
+
+		*)
+		PLUGIN_FULL_NAME=geoserver-${GEOSERVER_VERSION}-${PLUGIN_NAME}-plugin.zip
+		if [ "${TYPE}" == "ext" ]; then
+			NEWTYPE=extensions
+			PLUGIN_ARTIFACT_URL=${BASE_BUILD_URL_STABLE}/${GEOSERVER_VERSION}/${NEWTYPE}/${PLUGIN_FULL_NAME}
+		else
+			VERSION="${GEOSERVER_VERSION::-2}-SNAPSHOT"
+			PLUGIN_FULL_NAME=geoserver-${VERSION}-${PLUGIN_NAME}-plugin.zip
+			PLUGIN_ARTIFACT_URL=${BASE_BUILD_URL}/${GEOSERVER_VERSION::-2}.x/${TYPE}-latest/${PLUGIN_FULL_NAME}
+		fi
+		;;
+
+	esac
+
+
+  if [ ! -e "${PLUGIN_ARTIFACT_URL}" ]; then
+      mkdir -p "${PLUGIN_ARTIFACT_URL}"
+  fi
+
+
+
+  download_from_url_to_a_filepath "${PLUGIN_ARTIFACT_URL}" "${PLUGIN_ARTIFACT_DIRECTORY}/${PLUGIN_FULL_NAME}"
+}
+
+function download_fonts()  {
+    if [ ! -e "${FONTS_ARTIFACT_DIRECTORY}" ]; then
+        mkdir -p "${FONTS_ARTIFACT_DIRECTORY}"
+    fi
+    download_from_url_to_a_filepath "${EXTRA_FONTS_URL}" "${FONTS_ARTIFACT_DIRECTORY}/fonts.tar.gz"
+}
+
+function download_marlin()  {
```

Comment:
As Marlin renderer is built-in in Java 11 and this function is never called, it could be removed.

Reply:
I agree

Reply:
removed